tensorflow/lite/kernels/internal/reference/reference_ops.h

```diff
@@ -3326,57 +3326,46 @@ inline void PadImageStyle(const tflite::PadParams& op_params,
 }
 template <typename T>
 inline void StridedSlice(const tflite::StridedSliceParams& op_params,
-                         const RuntimeShape& unextended_input_shape,
+                         const RuntimeShape& input_shape,
                          const T* input_data,
-                         const RuntimeShape& unextended_output_shape,
+                         const RuntimeShape& output_shape,
                          T* output_data) {
-  // Note that the output_shape is not used herein.
-  tflite::StridedSliceParams params_copy = op_params;
+  const int slice_dimensions = input_shape.DimensionsCount();
+  T* out_ptr = output_data;
 
-  TFLITE_DCHECK_LE(unextended_input_shape.DimensionsCount(), 4);
-  TFLITE_DCHECK_LE(unextended_output_shape.DimensionsCount(), 4);
-  const RuntimeShape input_shape =
-      RuntimeShape::ExtendedShape(4, unextended_input_shape);
-  const RuntimeShape output_shape =
-      RuntimeShape::ExtendedShape(4, unextended_output_shape);
+  std::vector<int> start;
```

Comment:
I'm not a fan of heap allocating for every single invocation here. Can we avoid that?